homeassistant/components/sensor/recorder.py

```diff
@@ -363,6 +366,14 @@ def compile_statistics(  # noqa: C901
             entity_ids=entities_significant_history,
         )
         history_list = {**history_list, **_history_list}
+    # If there are no recent state changes, the sensor's state may already be pruned
+    # from the recorder. Get the state from the state machine instead.
+    for entity_id, _, _, _ in entities:
```

Comment:
We are fetching the state in `_get_entities`.  Should we store it there and use it here so we know its always there ?

Reply:
Absolutely, fixed in 5f4c44e